core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1394,9 +1394,10 @@ class KafkaApis(val requestChannel: RequestChannel,
       sendResponseMaybeThrottle(request, createResponse)
     }
 
-    if (!controller.isActive) {
+    if (!controller.isActive || !config.deleteTopicEnable) {
+      val error = if (!controller.isActive) Errors.NOT_CONTROLLER else Errors.NONE
```

Comment:
It can't be correct that the caller is left with the impression that the deletion completed successfully when, in fact, deletion is disabled and the topic still exists. Admittedly, none of the other existing errors looks like a good fit either. Perhaps there needs to be a new `Error.INCOMPATIBLE_WITH_CONFIG` or something for cases where the configuration prevents fulfilment of the request.

Reply:
I thought the same thing but I didn't include that error because, for example, the AdminManager swallows an exception in case the topic is already marked for deletion or even the AdminUtils writes in Zk the topic as deleted but the TopicDeletionManager will never do.

Anyway, I just added that new error with its corresponding exception :-)